### PR TITLE
Feature/golang pfconfig reload

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Create a pfconfig resource pool in golang in order to keep track of loaded resources so that they aren't loaded everytime

# Impacts
pfconfig in golang

# NEW Package(s) required
No

# Delete branch after merge
YES

# NEWS file entries
None required

### CODE REVIEW

File: go/pfconfigdriver/pool_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		t.Error("Resource wasn't loaded when calling a first time load")
+	}
+
+	expected = "pfdemo.org"
+	if gen.Domain != expected {
+		t.Errorf("Resource domain wasn't loaded correctly through resource pool. Got %s instead of %s", gen.Domain, expected)
+	}
+
+	// Test changing data in pfconfig and reloading the resource
+	cmd := exec.Command("sed", "-i.bak", "s/domain=pfdemo.org/domain=zammitcorp.com/g", "/usr/local/pf/t/data/pf.conf")
```
Comment (extrafu) [inline go/pfconfigdriver/pool_test.go:59]: hummm...

Comment (julsemaan) [reply go/pfconfigdriver/pool_test.go:59]: Needed a way to modify the config file without the packetfence perl libraries.

That was the best quick way I found

Don't worry, I restore the actual test file after the test

**Comment by julsemaan** [comment]
Please review + merge so I can migrate @fdurand code in #2029 to have dynamic reload of configuration

**Comment by julsemaan** [comment]
I changed the reload strategy and will open another PR with the pfsso stuff in the next days

Closing this

**Pull request closed without merging**